In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import os

# NHTSA CRSS Data

In [3]:
maneuver_2017 = pd.read_csv('data/CRSS/2017/MANEUVER.CSV')

In [4]:
maneuver_2017.shape

(97628, 11)

In [5]:
maneuver_2017.head()

CASENUM  VEH_NO  REGION  PSU   PJ  PSU_VAR  URBANICITY  STRATUM  \
0  201700000117       1       4   64  305       64           1        3   
1  201700000117       2       4   64  305       64           1        3   
2  201700000119       1       4   64  305       64           1        6   
3  201700000119       2       4   64  305       64           1        6   
4  201700000119       3       4   64  305       64           1        6   

   MDRMANAV  PSUSTRAT     WEIGHT  
0        98        23  29.270311  
1        98        23  29.270311  
2        92        23  62.100147  
3         4        23  62.100147  
4         4        23  62.100147

In [6]:
maneuver_2017[maneuver_2017.MDRMANAV==3].shape

(435, 11)

There's only 435 instances of maneuvers for all animals. Fuck

In [11]:
accident_2017 = pd.read_csv('data/CRSS/2017/ACCIDENT.CSV')
vehicle_2017 = pd.read_csv('data/CRSS/2017/VEHICLE.CSV')

/home/jeremy/.anaconda3/envs/inaturalist/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (26,27,74) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [8]:
accident_2017[accident_2017.HARM_EV==7].shape

(35, 51)

In [9]:
accident_2017.shape

(54969, 51)

In [12]:
vehicle_2017[vehicle_2017.M_HARM==7].shape

(23, 87)

# FARS Data

In [13]:
maneuver_2017 = pd.read_csv('data/FARS/2017/Maneuver.csv')

In [14]:
maneuver_2017.shape

(52656, 4)

In [15]:
maneuver_2017.head()

STATE  ST_CASE  VEH_NO  MDRMANAV
0      1    10001       1        98
1      1    10002       1        98
2      1    10003       1        98
3      1    10003       2        98
4      1    10003       3        98

In [16]:
maneuver_2017[maneuver_2017.MDRMANAV==3].shape

(118, 4)

There's only 435 instances of maneuvers for all animals. Fuck

In [11]:
accident_2017 = pd.read_csv('data/CRSS/2017/ACCIDENT.CSV')
vehicle_2017 = pd.read_csv('data/CRSS/2017/VEHICLE.CSV')

/home/jeremy/.anaconda3/envs/inaturalist/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (26,27,74) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [8]:
accident_2017[accident_2017.HARM_EV==7].shape

(35, 51)

In [9]:
accident_2017.shape

(54969, 51)

In [12]:
vehicle_2017[vehicle_2017.M_HARM==7].shape

(23, 87)